frontend/src/components/custom/QuestionTable/data-table.tsx

```diff
@@ -68,6 +70,32 @@ export function DataTable<TData, TValue>({
     table.getColumn("topics")?.setFilterValue(selectedValues);
   };
 
+  const [isAdmin, setIsAdmin] = useState<Boolean>(false);
+  
+  async function fetchStatus(): Promise<Boolean> {
+
+      const res = await fetchAdminStatus();
```

Comment:
could the `callUserFunction` be called here directly instead ? im not sure why it needs to route to `fetchAdminStatus` from UserFunctions

Reply:
Was following the precedent set by the way the questions were fetched. Basically components in the frontend will use functions set in services/UserFunctions.ts to get information from the backend, and the functions in UserFunctions.ts go through the lib/utils.ts which is shared by different functions.

By right the neater way is probably to refactor `callFunction` and `callUserFunction` into the same function and include logic to send the requests to the correct backend service. I'll work on that next.